ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -301,83 +153,91 @@ export class ImportedDocuments extends React.Component {
     });
   };
 
-  renderCancelButton = () => {
-    const { isLoading } = this.state;
-    const { taskId } = this.props;
-
-    return (
-      isLoading && (
-        <>
-          <CancelImportTask logId={taskId} />{' '}
-          <Icon loading name="circle notch" />
-          This may take a while. You may leave the page, the process will
-          continue in background. ImportedSearch
-        </>
-      )
-    );
-  };
-
   render() {
+    const { data, taskId, isLoading, error } = this.props;
     const {
-      data,
       statistics,
       selectedResult,
       filteredRecords,
       activePage,
     } = this.state;
+
+    const dataAvailable = !_isEmpty(data);
+    const filteredRecordsAvailable = !_isEmpty(filteredRecords);
+    const filteredRecordsNotNull = !_isNull(filteredRecords);
```

Comment:
This is curious. Why not simply have one variable `const filteredRecordsAvailable = !_isEmpty(filteredRecords);`
and then check if there are no results with `!filteredRecordsAvailable`?

Reply:
In most cases i just tried to preserve the condition logic that was in place because without having worked on it is hard to predict why that condition was placed there and what it might break.